src/components/index.js

```diff
@@ -367,6 +372,25 @@ class Joyride extends React.Component {
     }
   };
 
+  shouldScroll = (
+    disableScrolling,
+    index,
+    scrollToFirstStep,
+    lifecycle,
+    step,
+    target,
+    prevState,
+  ) => {
+    return (
+      !disableScrolling &&
+      (index !== 0 || (scrollToFirstStep || lifecycle === LIFECYCLE.TOOLTIP)) &&
+      step.placement !== 'center' &&
+      (!step.isFixed || !hasPosition(target)) && // fixed steps don't need to scroll
+      (prevState.lifecycle !== lifecycle &&
+        [LIFECYCLE.BEACON, LIFECYCLE.TOOLTIP].includes(lifecycle))
+    );
+  };
```

Comment:
Why would you want to extract this? I might have missed the point here but it looks like it didn't change much what was there beforehand right?

Reply:
because codeclimate failed this part.

I forgot the error message.but I think codeclimate says expression is too long.

Reply:
Aah okay I see, sorry on my part I wasn't aware!